services/src/main/java/io/grpc/services/MetricRecorder.java

```diff
@@ -40,21 +40,21 @@ private MetricRecorder() {}
   /**
    * Update the metrics value corresponding to the specified key.
    */
-  public void setUtilizationMetric(String key, double value) {
+  public void putUtilizationMetric(String key, double value) {
     metricsData.put(key, value);
   }
 
   /**
    * Replace the whole metrics data using the specified map.
    */
-  public void setAllUtilizationMetrics(Map<String, Double> metrics) {
+  public void putAllUtilizationMetrics(Map<String, Double> metrics) {
```

Comment:
"put" wouldn't remove entries missing from `metrics`. The Map API doesn't have an equivalent for this. Either this could remain "set" or we'd change its behavior to not clear the existing entries (at which point it doesn't serve much of a purpose and could be deleted, probably).